flow360/component/project.py

```diff
@@ -0,0 +1,563 @@
+        target : AssetOrResource
+            The target asset or resource to run the simulation against.
+        draft_name : str, optional
+            The name of the draft to create for the simulation run (default is None).
+        fork : bool, optional
+            Indicates if the simulation should fork the existing case (default is False).
+        run_async : bool, optional
+            Specifies whether the simulation should run asynchronously (default is True).
+
+        Returns
+        -------
+        AssetOrResource
+            The destination asset
+
+        Raises
+        ------
+        Flow360ValueError
+            If the simulation parameters lack required length unit information, or if the
+            root asset (Geometry or VolumeMesh) is not initialized.
+        """
+        defaults = self.get_simulation_json()
+
+        cache_key = "private_attribute_asset_cache"
+        length_key = "project_length_unit"
+
+        if cache_key not in defaults:
+            if length_key not in defaults[cache_key]:
+                raise Flow360ValueError("Simulation params do not contain default length unit info")
+
+        length_unit = defaults[cache_key][length_key]
+
+        with model_attribute_unlock(params.private_attribute_asset_cache, length_key):
+            params.private_attribute_asset_cache.project_length_unit = LengthType.validate(
+                length_unit
+            )
+
+        root_asset = None
+        root_type = self.metadata.root_item_type
+
+        if root_type == RootType.GEOMETRY:
+            root_asset = self._geometry
+        elif root_type == RootType.VOLUME_MESH:
+            root_asset = self._volume_mesh
+
+        if not root_asset:
+            raise Flow360ValueError("Root asset is not initialized")
+
+        draft = Draft.create(
+            name=draft_name,
+            project_id=self.metadata.id,
+            source_item_id=self.metadata.root_item_id,
+            source_item_type=root_type.value,
+            solver_version=self.solver_version,
+            fork_case=fork,
+        ).submit()
+
+        entity_info = root_asset.entity_info
+        registry = params.used_entity_registry
+        old_draft_entities = entity_info.draft_entities
+        for _, old_entity in enumerate(old_draft_entities):
+            try:
+                registry.find_by_naming_pattern(old_entity.name)
+            except ValueError:
+                continue
+
+        with model_attribute_unlock(params.private_attribute_asset_cache, "project_entity_info"):
+            params.private_attribute_asset_cache.project_entity_info = entity_info
+
+        draft.update_simulation_params(params)
+        destination_id = draft.run_up_to_target_asset(target)
+
+        RestApi(ProjectInterface.endpoint, id=self.metadata.id).patch(
+            json={
+                "lastOpenItemId": destination_id,
+                "lastOpenItemType": target.__name__,
+            }
+        )
```

Comment:
initialise this in constructor and run:
```suggestion
        self.webapi.patch(
            json={
                "lastOpenItemId": destination_id,
                "lastOpenItemType": target.__name__,
            }
        )
```

Reply:
Fixed in [9ba5587](https://github.com/flexcompute/Flow360/pull/511/commits/9ba5587bee8a37191788f1094fe8503dd5cda41d)